### PR TITLE
Bump Razor and Roslyn to respond to Razor serialization changes

### PR DESCRIPTION
**Pull request by davidwengier**
Razor going from 87b4727a3e721a1766a759133fcf35333648a143 to fd837e1f2df38690099059b0a315ed6eadd6d472.
Diff: https://github.com/dotnet/razor/compare/87b4727a3e721a1766a759133fcf35333648a143...fd837e1f2df38690099059b0a315ed6eadd6d472

### CODE REVIEW

**Comment by davidwengier** [comment]
Forgot to run `updateRoslynVersion` too 🤦‍♂️

**Review by dibarbet — APPROVED** [review]
If you're planning on updating Razor again later this week feel free to ignore the changelog update for now.  Roslyn one can be ignored (will be updated again soon)

File: CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,11 @@
 * Renamed settings (PR: [#7356](https://github.com/dotnet/vscode-csharp/pull/7356))
   * `dotnet.implementType.insertionBehavior` to `dotnet.typeMembers.memberInsertionLocation`
   * `dotnet.implementType.propertyGenerationBehavior` to `dotnet.typeMembers.propertyGenerationBehavior`
+* Update Razor to 9.0.0-preview.24366.2 (PR: [#7384](https://github.com/dotnet/vscode-csharp/pull/7384))
```
Comment (dibarbet) [inline CHANGELOG.md:18]: This needs to be in a new section now (and this one renamed to 2.40.x

**Review by dibarbet — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (davidwengier) [reply CHANGELOG.md]: Thank you. I alway get this wrong

**Review by ryzngard — APPROVED** [review]

**Review by dibarbet — APPROVED** [review]

**Pull request merged**